assets/src/design-system/components/typography/text/index.js

```diff
@@ -26,36 +26,63 @@ import PropTypes from 'prop-types';
 import { THEME_CONSTANTS, themeHelpers } from '../../../theme';
 import { defaultTypographyStyle } from '../styles';
 
-export const Text = styled.p`
+const textCss = ({ isBold, size, theme }) => css`
   ${defaultTypographyStyle}
-  ${({ as, isBold, size, theme }) => {
-    const asLink =
-      as === 'a' &&
-      css`
-        color: ${theme.colors.fg.linkNormal};
-        text-decoration: none;
-        cursor: pointer;
-        &:hover {
-          color: ${theme.colors.fg.linkHover};
-        }
+  ${themeHelpers.expandPresetStyles({
+    preset: theme.typography.presets.paragraph[size],
+    theme,
+  })};
+  font-weight: ${isBold
+    ? theme.typography.weight.bold
+    : theme.typography.presets.paragraph[size].weight};
+`;
+
+const Paragraph = styled.p`
+  ${textCss};
+`;
+
+const Span = styled.span`
+  ${textCss};
+`;
+
+const Label = styled.label`
+  ${textCss};
 
-        ${themeHelpers.focusableOutlineCSS(theme.colors.border.focus)}
-      `;
-    return css`
-      ${themeHelpers.expandPresetStyles({
-        preset: theme.typography.presets.paragraph[size],
-        theme,
-      })};
-      font-weight: ${isBold
-        ? theme.typography.weight.bold
-        : theme.typography.presets.paragraph[size].weight};
-      ${asLink};
-    `;
-  }}
+  color: ${({ disabled, theme }) =>
+    disabled ? theme.colors.fg.disable : 'auto'};
 `;
 
+const Anchor = styled.a`
+  ${({ theme }) => css`
+    color: ${theme.colors.fg.linkNormal};
+    text-decoration: none;
+    cursor: pointer;
+
+    :hover {
+      color: ${theme.colors.fg.linkHover};
+    }
+
+    ${themeHelpers.focusableOutlineCSS(theme.colors.border.focus)}
+  `};
+`;
+
+export const Text = ({ as, disabled, ...props }) => {
+  switch (as) {
+    case 'a':
+      return <Anchor {...props} />;
+    case 'label':
+      return <Label disabled={disabled} {...props} />;
+    case 'span':
+      return <Span {...props} />;
+    default:
+      return <Paragraph {...props} />;
+  }
+};
```

Comment:
I like this a lot, great update.

Reply:
Something to consider when using `<Text as="a"` instead of `Anchor` is that the linter won't detect those as actual links.

See https://github.com/google/web-stories-wp/blob/21d3117d9db658935c018cfdbc2ba2f1617d0649/.eslintrc#L232-L239

Reply:
Oh interesting, I'll update that to use our `<Link />` component

Reply:
`Link` is now a named export from the design system:
<image src="https://user-images.githubusercontent.com/22185279/106494546-cd1c4580-6477-11eb-9be5-3f65f68d6201.png" width="300px" />

Reply:
@samwhale That's good, but still if someone uses `<Text as="a"` somewhere, the linter won't catch it.

Reply:
@swissspidy I refactored it out of the `<Text />` component. It's a standalone component now ✨